macos/FluentUI/AvatarView/AvatarView.swift

```diff
@@ -409,22 +429,34 @@ open class AvatarView: NSView {
 		return initials?.localizedUppercase
 	}
 
-	/// Returns a color table index for a given display name
-	/// - parameter identifyingString: the unique identifying string for the user
-	///
-	/// - Returns: a unique index generated by walking backwards through the letters in the \p identifyingString
-	/// creating a cumulative hash
-	///
-	/// - note: the returned index is not limited to any size and should not be considered safe for indexing into an array without
-	/// first checking the size of the array in question
-	@objc(colorIndexForIdentifyingString:)
-	public static func colorIndex(for identifyingString: String) -> Int {
-		return identifyingString.utf16.enumerated().reversed().reduce(0) { (hashCode, enumeratedCodePoints) -> Int in
-			// upcast our UInt16s to standard Ints as that's how the hashing algorithm works in other codebases
-			let integerCodePoint = Int(enumeratedCodePoints.element)
-			let shift = enumeratedCodePoints.offset % 8
-			return hashCode ^ Int((integerCodePoint << shift) + (integerCodePoint >> (8 - shift)))
+	@objc(getInitialsColorSetFromPrimaryText:secondaryText:)
+	public static func getInitialsColorSet(fromPrimaryText primaryText: String?, secondaryText: String?) -> ColorSet {
+		// Set the color based on the primary text and secondary text
+		var combined: String
```

Comment:
does this has to be var instead of let?

Reply:
its mutated on line 439 and 441 so this needs to be a var. If you are referring to initializing this one based on condition then yes we can. Since the condition to initialize this is complex I am concerned that would make code readability much harder. Is there any other specific reason you would to make it immutable?

Reply:
the recommendation is always let unless there is a reason to be var in swift. Isn't combined only set once depending on which if statement ?

Reply:
if `combined` is defined as a `let` variable then it won't be accessible outside of its enclosing if/else. That's the only reason for using `var`

Reply:
```swift
let combined: String = {
  //... logic
}()
```

Reply:
^ I think that works. I feel like I've used it at least once

Reply:
Sync'ed with Saad offline the above conditional initialization didn't work. @Harrie you have any other ideas?

Reply:
@kubalani if you just change it to let it'll work because you're initializing it in one of the 3 paths of the if statement. And it's not changed afterwards.

```
                let combined: String
		if let secondaryText = secondaryText, let primaryText = primaryText, secondaryText.count > 0 {
			combined = primaryText + secondaryText
		} else if let primaryText = primaryText {
			combined = primaryText
		} else {
			combined = ""
		}
```

Alternatively you can also do the closure initialization (as @Saadnajmi suggested):
Not sure what didn't work there?

```
		let combined: String = {
			if let primaryText = primaryText {
				if let secondaryText = secondaryText, secondaryText.count > 0 {
					return primaryText + secondaryText
				}

				return primaryText
			}

			return ""
		}()
```

Reply:
just replacing `let` with `var` did worked. Swift doesn't like the second initialization

Reply:
Swift does not allow a second assignment because "let" define a constant that can only be initialized (first assignment).
You can still declare the constant and initialize it later.
Swift will also catch if you try to use a constant before it's initialized.